Cubical/Data/FinSet/Properties.agda

```diff
@@ -0,0 +1,253 @@
+{-
+
+This file contains:
+
+- Some useful implications;
+
+- Closure properties of FinSet under several type constructors.
+
+-}
+{-# OPTIONS --safe #-}
+
+module Cubical.Data.FinSet.Properties where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Equiv
+
+open import Cubical.HITs.PropositionalTruncation renaming (rec to TruncRec)
+
+open import Cubical.Data.Nat
+open import Cubical.Data.Unit
+open import Cubical.Data.Empty renaming (rec to EmptyRec)
+open import Cubical.Data.Sum
+open import Cubical.Data.Sigma
+
+open import Cubical.Data.Fin
+open import Cubical.Data.SumFin renaming (Fin to SumFin) hiding (discreteFin)
+open import Cubical.Data.FinSet hiding (isFinSetΣ)
+open import Cubical.Data.FinSet.SumFinReduction
+open import Cubical.Data.FinSet.FiniteChoice
+
+open import Cubical.Relation.Nullary
+open import Cubical.Relation.Nullary.DecidableEq
+open import Cubical.Relation.Nullary.HLevels
+
+private
+  variable
+    ℓ ℓ' ℓ'' ℓ''' : Level
+
+isFinSet→Discrete : {X : Type ℓ} → isFinSet X → Discrete X
+isFinSet→Discrete p =
+  TruncRec isPropDiscrete (λ (_ , p) → EquivPresDiscrete (invEquiv p) discreteFin) p
+
+isFinSet→isSet : {X : Type ℓ} → isFinSet X → isSet X
+isFinSet→isSet p = Discrete→isSet (isFinSet→Discrete p)
```

Comment:
Wouldn't it be slicker to prove this from `isProp-isFinSet` ?

Reply:
This proof is roundabout actually. But how to use `isProp-isFinSet`?

Reply:
Can't you just use https://github.com/agda/cubical/blob/master/Cubical/Foundations/Prelude.agda#L462 ?

Reply:
Maybe we should upstream this to the Base file as it's what motivates calling these `isFinSet` instead of just `isFinite`